services/core-api/app/api/mines/explosives_permit_amendment/models/explosives_permit_amendment.py

```diff
@@ -162,12 +221,120 @@ def update(self,
+                explosives_permit_amendment_doc = ExplosivesPermitAmendmentDocumentXref(
+                    mine_document_guid=mine_doc.mine_document_guid,
+                    explosives_permit_amendment_id=self.explosives_permit_id,
+                    explosives_permit_amendment_document_type_code=explosives_permit_amendment_document_type_code)
+                explosives_permit_amendment_doc.mine_document = mine_doc
+                self.documents.append(explosives_permit_amendment_doc)
+
+        # Check for application status changes or application is Approved to regenerate permit and letter.
         if application_status:
             if application_status != 'REC' and application_status != 'APP':
                 self.decision_timestamp = datetime.utcnow()
                 self.decision_reason = decision_reason
 
+            if (self.application_status == 'REC'
+                    or self.application_status == 'APP') and application_status == 'APP':
+                from app.api.document_generation.resources.explosives_permit_amendment_document_resource import ExplosivesPermitAmendmentDocumentResource
+                from app.api.mines.explosives_permit.resources.explosives_permit_document_type import ExplosivesPermitDocumentGenerateResource
+
+                def create_permit_enclosed_letter():
+                    mine = self.mine
+                    # TODO: Implement a method in the document type to automatically get all read-only context values.
+                    template_data = {
+                        'letter_date': letter_date,
+                        'letter_body': letter_body,
+                        'rc_office_email': mine.region.regional_contact_office.email,
+                        'rc_office_phone_number': mine.region.regional_contact_office.phone_number,
+                        'rc_office_fax_number': mine.region.regional_contact_office.fax_number,
+                        'rc_office_mailing_address_line_1':
+                        mine.region.regional_contact_office.mailing_address_line_1,
+                        'rc_office_mailing_address_line_2':
+                        mine.region.regional_contact_office.mailing_address_line_2,
+                        'is_draft': False
+                    }
+                    explosives_permit_amendment_document_type = ExplosivesPermitDocumentType.get_with_context(
+                        'LET', self.explosives_permit_amendment_guid)
+                    template_data = explosives_permit_amendment_document_type.transform_template_data(
+                        template_data, self)
+                    token = ExplosivesPermitDocumentGenerateResource.get_explosives_document_generate_token(
+                        explosives_permit_amendment_document_type.explosives_permit_document_type_code,
+                        self.explosives_permit_amendment_guid, template_data)
+                    # TODO: Remove Logs for generate document
```

Comment:
You've got a couple TODOs here that look like they were left in by mistake, but all in all looks great! I like the Mixin stuff.